File: calculator.cpp
#include <iostream>
#include <stdexcept>

using namespace std;

void add(double num1, double num2) {
    cout << "Result: " << num1 + num2 << endl;
}

void subtract(double num1, double num2) {
    cout << "Result: " << num1 - num2 << endl;
}

void multiply(double num1, double num2) {
    cout << "Result: " << num1 * num2 << endl;
}

void divide(double num1, double num2) {
    if (num2 == 0) {
        throw runtime_error("Error! Division by zero.");
    } else {
        cout << "Result: " << num1 / num2 << endl;
    }
}

int main() {
    int choice;
    double num1, num2;

    cout << "Select operation:" << endl;
    cout << "1. Add" << endl;
    cout << "2. Subtract" << endl;
    cout << "3. Multiply" << endl;
    cout << "4. Divide" << endl;

    cout << "Enter choice (1/2/3/4): ";
    cin >> choice;

    if (cin.fail() || choice < 1 || choice > 4) {
        cout << "Invalid input" << endl;
        return 1;
    }

    cout << "Enter first number: ";
    cin >> num1;

    if (cin.fail()) {
        cout << "Invalid input" << endl;
        return 1;
    }

    cout << "Enter second number: ";
    cin >> num2;

    if (cin.fail()) {
        cout << "Invalid input" << endl;
        return 1;
    }

    try {
        switch (choice) {
            case 1:
                add(num1, num2);
                break;
            case 2:
                subtract(num1, num2);
                break;
            case 3:
                multiply(num1, num2);
                break;
            case 4:
                divide(num1, num2);
                break;
            default:
                cout << "Invalid choice" << endl;
        }
    } catch (const runtime_error& e) {
        cout << e.what() << endl;
    }

    return 0;
}

File: calculator.py
import tkinter as tk
from tkinter import messagebox

def add():
    try:
        num1 = float(entry1.get())
        num2 = float(entry2.get())
        result = num1 + num2
        result_var.set(result)
    except ValueError:
        messagebox.showerror("Input Error", "Please enter valid numbers")

def subtract():
    try:
        num1 = float(entry1.get())
        num2 = float(entry2.get())
        result = num1 - num2
        result_var.set(result)
    except ValueError:
        messagebox.showerror("Input Error", "Please enter valid numbers")

def multiply():
    try:
        num1 = float(entry1.get())
        num2 = float(entry2.get())
        result = num1 * num2
        result_var.set(result)
    except ValueError:
        messagebox.showerror("Input Error", "Please enter valid numbers")

def divide():
    try:
        num1 = float(entry1.get())
        num2 = float(entry2.get())
        if num2 == 0:
            messagebox.showerror("Math Error", "Division by zero is not allowed")
            result_var.set("Error")
        else:
            result = num1 / num2
            result_var.set(result)
    except ValueError:
        messagebox.showerror("Input Error", "Please enter valid numbers")

# Create the main window
root = tk.Tk()
root.title("Simple Calculator")

# Create and place widgets
tk.Label(root, text="Number 1:").grid(row=0, column=0, padx=10, pady=10)
tk.Label(root, text="Number 2:").grid(row=1, column=0, padx=10, pady=10)

entry1 = tk.Entry(root)
entry2 = tk.Entry(root)
entry1.grid(row=0, column=1, padx=10, pady=10)
entry2.grid(row=1, column=1, padx=10, pady=10)

tk.Button(root, text="Add", command=add).grid(row=2, column=0, padx=10, pady=10)
tk.Button(root, text="Subtract", command=subtract).grid(row=2, column=1, padx=10, pady=10)
tk.Button(root, text="Multiply", command=multiply).grid(row=3, column=0, padx=10, pady=10)
tk.Button(root, text="Divide", command=divide).grid(row=3, column=1, padx=10, pady=10)

result_var = tk.StringVar()
tk.Label(root, text="Result:").grid(row=4, column=0, padx=10, pady=10)
tk.Label(root, textvariable=result_var).grid(row=4, column=1, padx=10, pady=10)

# Run the application
root.mainloop()
